mcnp_input_reader.py

```diff
@@ -223,29 +438,185 @@ def cells_with_mat(mat_num, cell_list):
     return cells
 
 
-def print_cell(cell):
-    """ pretty printing of cell object """
-    print("Cell number: ", cell.number)
-    print("Cell material: ", cell.mat)
-    print("Cell density: ", cell.density)
-    print("Cell geom: ", cell.geom)
-    print("Cell surfaces: ", cell.surfaces)
-    print("Cell imp p:", cell.imp_p)
-    print("Cell imp n:", cell.imp_n)
+def float_check(n):

```

Comment:
need to think about this some more - i think it is best practice to only catch the specific exception not all exceptions and i suspect there is a better way to do this than this try except using isinstance(x, float) where this is called

Reply:
Had to settle for float() because the list is a list of strings so isinstance just sees numbers as strings, but float() attempts to convert to float, which isnt possible for the character type entries. Did narrow down the exception though

Reply:
perhaps isnumeric() is the right thing to use then

Reply:
tried isnumeric but it wont handle decimal point or negative value cases